src/StackExchange.Redis/ServerEndPoint.cs

```diff
@@ -576,7 +576,7 @@ internal Message GetTracerMessage(bool assertIdentity)
             var map = Multiplexer.CommandMap;
             Message msg;
             const CommandFlags flags = CommandFlags.NoRedirect | CommandFlags.FireAndForget;
-            if (assertIdentity && map.IsAvailable(RedisCommand.ECHO))
+            if (checkResponse && map.IsAvailable(RedisCommand.ECHO))
             {
                 msg = Message.Create(-1, flags, RedisCommand.ECHO, (RedisValue)Multiplexer.UniqueId);
```

Comment:
When this `msg` is sent by `KeepAlive()` its result is processed by `ResultProcessor.Tracer`, and I don't think `ResultProcessor.Tracer` actually checks the result for `Multiplexer.UniqueId`. But its buddy `ResultProcessor.EstablishConnection` _would_ check the result thanks to this check in `TracerProcessor.SetResultCore()`: 
`(!establishConnection || result.IsEqual(connection.BridgeCouldBeNull?.Multiplexer?.UniqueId))`
(in `ResultProcessor.Tracer`, `!establishConnection` is always true)

Not sure what the right fix would be, but maybe add a flag to force `ResultProcessor.Tracer` to check the result?

Reply:
We looked here - are good since we're on the `PING` path instead of `ECHO` meaning we'll check for `PONG` on the response.